File: Python/ObserverPattern/NewsAgency/engineering_news_agency.py
from NewsAgency.news_agency import NewsAgency


class EngineeringNewsAgency(NewsAgency):
    __instance = None

    @staticmethod
    def get_instance():
        if EngineeringNewsAgency.__instance is None:
            EngineeringNewsAgency()
        return EngineeringNewsAgency.__instance

    def __init__(self):
        if EngineeringNewsAgency.__instance is not None:
            raise Exception("This class is singleton")
        else:
            EngineeringNewsAgency.__instance = self
            self.publisher = "Engineering News Agency"

File: Observer/Java/NewsAgency/src/com/laxmena/EngineeringNewsAgency.java
package com.laxmena;

import java.time.LocalDateTime;

public class EngineeringNewsAgency extends NewsAgency {
    private static EngineeringNewsAgency instance = new EngineeringNewsAgency();
    private String PUBLISHER_NAME = "Engineering News";

    private void EngineeringNewsAgency() {
        this.name = "Engineering News Agency";
    }

    public static EngineeringNewsAgency getInstance() {
        return instance;
    }

    public void addNewNews(String title, String content) {
        LocalDateTime publishedOn = LocalDateTime.now();
        News news = new News(title, content, publishedOn, PUBLISHER_NAME);
        this.newsList.add(news);
        // Whenever there is a new news, all the subscribers should be updated
        notifySubscribers();
    }
}
